### DISCUSSION TITLE
Does BERTopic come with any methods that assist user to determine the number of topic (k)?

### DISCUSSION THREAD

**Discussion by chentitus** [MaartenGr/BERTopic#1235] [General]
Dear Maarten,

I am pretty new to Python and BERTopic. I wonder if BERTopic has any methods that can help users decide the number of topic/clusters, such as perplexity score. Any help is appreciated!

**Comment by MaartenGr** [comment]
Choosing the number of topics highly depends on the underlying clustering algorithm as each will behave differently. For instance, k-Means with a centroid-based technique handles clustering differently from a density-based algorithm like HDBSCAN. So it would depend on the underlying model that you use. I personally would advise involving human evaluation as much as possible as grid-searching evaluation metrics often leads to misleading results.
